services/sources/publisher_backlist_service.py

```diff
@@ -25,17 +27,20 @@ def get_records(
         limit: Optional[int]=None
     ) -> list[PublisherBacklistMapping]:
         array_json_records = self.get_records_json(full_import, start_timestamp, offset, limit)
-
+        complete_records = []
         for json_dict in array_json_records:
             for records_value in json_dict['records']:
                 try:
                     record_metadata_dict = records_value['fields']
-                    record = PublisherBacklistMapping(record_metadata_dict)
-                    record.applyMapping()
+                    pub_backlist_record = PublisherBacklistMapping(record_metadata_dict)
+                    pub_backlist_record.applyMapping()
+                    self.addHasPartMapping(pub_backlist_record, pub_backlist_record.record)
+                    self.storePDFManifest(pub_backlist_record.record)
+                    complete_records.append(pub_backlist_record)
                 except Exception:
                     logger.exception(f'Failed to process Publisher Backlist record')
```

Comment:
I recommend adding the record_value at the very least in this exception message so we know which record failed.

Reply:
I can do that